src/test/ui/consts/const-eval/issue-49296.stderr

```diff
@@ -4,7 +4,7 @@ error: any use of this value will cause an error
 LL | const X: u64 = *wat(42);
    | ---------------^^^^^^^^-
    |                |
-   |                dangling pointer was dereferenced
+   |                pointer to alloc2 was dereferenced after this allocation got freed
```

Comment:
Exposing alloc IDs to users seems suboptimal

Reply:
Note that other errors exposed alloc IDs already, now it's just more of them.
For Miri I actually find this quite helpful as you can then re-run with "track alloc ID" to figure out when it got deallocated. Maybe there is a way to use formatting flags (like `{:?}` vs `{:#?}`) to switch between two different outputs modes for errors?

Reply:
Let's use it as pressure to make debugging const eval memory possible for users